### ISSUE TITLE
[combobox][autocomplete] `aria-readonly` on `Clear` and `Popup` components when `readOnly`

### ISSUE THREAD

**Issue by ZeeshanTamboli** [mui/base-ui#3906]
# Bug report

## Current behavior

`aria-readonly` is added on `Clear` and `Popup` components for Autocomplete and Combobox when `readOnly` is used on `Root`.

## Expected behavior

`aria-readonly` seems unnecessary on `Clear` and `Popup` components for Autocomplete and Combobox. They are rendered as buttons.

## Reproducible example

StackBlitz: https://stackblitz.com/edit/3hx98nwo?file=src%2FApp.tsx

## Base UI version

v1.1.0

## Which browser are you using?

Chrome

## Which OS are you using?

Windows

## Which assistive tech are you using (if applicable)?

NVDA

## Additional context

Nothing is announced on NVDA screen-reader when `aria-readonly="true"`. They are rendered as buttons and `aria-readonly` is not applicable to them, unless I am missing something. 

`readOnly` docs: https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes/readonly
`aria-readonly` docs: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-readonly
